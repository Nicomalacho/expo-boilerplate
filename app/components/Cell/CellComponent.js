import React from 'react';
import { View, Image, StyleSheet, Text } from 'react-native';

export default class Cell extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            source={{ uri: this.props.item.thumbnailUrl }}
          />
        </View>
        <View>
          <Text>{this.props.item.title}</Text>
          <Text>{this.props.item.id}</Text>
        </View>
      </View>
    )
  }
}


const styles = StyleSheet.create({
  container: {
    padding: 10,
    fontSize: 18,
    height: 120,
    flexDirection: 'row'
  },
  image: {
    width: 100, 
    height: 100,
    borderRadius: 50
  }
})

