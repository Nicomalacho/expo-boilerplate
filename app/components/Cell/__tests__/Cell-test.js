// __tests__/Dummy-test.js
import React from 'react';
import Cell from '../CellComponent';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const item = {
    id: '1',
    title: 'some title to test',
    thumbnailUrl: 'https://via.placeholder.com/150/771796'
  }
  const tree = renderer.create(<Cell item={item}/>).toJSON();
  expect(tree).toMatchSnapshot();
});