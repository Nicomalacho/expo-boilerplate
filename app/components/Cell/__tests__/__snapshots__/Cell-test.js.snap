// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "fontSize": 18,
      "height": 120,
      "padding": 10,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "https://via.placeholder.com/150/771796",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View>
    <Text>
      some title to test
    </Text>
    <Text>
      1
    </Text>
  </View>
</View>
`;
